### ISSUE TITLE
Support for REUSE Specification (License Detection)

### ISSUE THREAD

**Issue by n0toose** [go-gitea/gitea#28672]
### Feature Description

I think it would be great if Gitea was able to detect multiple licenses under a `LICENSES` folder based on the SPDX identifier included in the filenames of present in that folder, as described by the Free Software Foundation Europe's REUSE Specification. (https://reuse.software)

That way, it would be possible to include multiple licenses easily without looking for the SPDX headers in every file that is present in the repository, while retaining the ability to include licensing information for binary files.

A "lazier" implementation that does not check for full compliance, but just looks at the contents of the `LICENSES` folder (unlike re-implementing or somehow integrating https://github.com/fsfe/reuse-tool - which does check every file) could work and that repositories using it could reduce some computational overhead (w.r.t. https://github.com/go-gitea/gitea/pull/24872#discussion_r1410524792) at scale.

I believe that the benefits and overall simplicity of supporting such an extension are highlighted in the review process of this pull request: https://github.com/go-gitea/gitea/pull/24872

### Screenshots

_Not applicable_

**Comment by n0toose** [comment]
My motivation for this issue is also the fact that our project is pivoting towards utilizing that standard for its own projects and wanting to recommend it further to other free and open source developers.

I'd be interested in taking a look, but I already have a few outstanding PRs to circle back to and there is an actively developed PR that changes these very components.

**Comment by wxiaoguang** [comment]
License detection is supported.

**Comment by mcepl** [comment]
No, it doesn’t … please compare (otherwise identical) https://git.sr.ht/~mcepl/ollama_dynamic_models (where Sourcehut correctly recognises that the project is under Apache-2.0 license) and https://gitea.com/mcepl/ollama_dynamic_models (where I don’t see any indication of the license used anywhere).

Please, reopen.

**Comment by wxiaoguang** [comment]
Hmm, at the moment, IIRC it only detects the license files in the root directory.

Maybe it's not difficult to add more candidate paths to detect.
